src/Workleap.Extensions.OpenAPI.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -168,16 +173,45 @@ public void ValidateEndpointResponseWithAnnotationType(SymbolAnalysisContext con
                 }
             }
 
-            // For the annotations [SwaggerResponse(<StatusCode>, "description", <typeof()>]
+            // For the annotations [SwaggerResponse(<StatusCode>, "description", <typeof()>] and [SwaggerResponse(<StatusCode>, "description", Type = <typeof()>]
             else if (attribute.AttributeClass.ConstructedFrom.Equals(this.SwaggerResponseSymbol, SymbolEqualityComparer.Default))
             {
-                if (attribute.ConstructorArguments.Length > 2 && attribute.ConstructorArguments[0].Value is int statusCodeValue && attribute.ConstructorArguments[2].Value is ITypeSymbol type)
+                if (attribute.NamedArguments.Length > 0 && ValidateTypeInNamedArguments(context, attribute, methodSignatureStatusCodeToTypeMap))
```

Comment:
Is this check redundant: attribute.NamedArguments.Length > 0, I think it could be remove and work correctly by just checking the method ValidateTypeInNamedArguments?